src/main/java/carpet/script/CarpetScriptServer.java

```diff
@@ -150,17 +185,20 @@ public ScriptHost getHostByName(String name)
         return modules.get(name);
     }
 
-    public boolean addScriptHost(ServerCommandSource source, String name, boolean perPlayer, boolean autoload)
+    public boolean addScriptHost(ServerCommandSource source, String name, boolean perPlayer, boolean autoload, boolean isRuleApp)
     {
         //TODO add per player modules to support player actions better on a server
+        if (!worldInitialized) return false;
```

Comment:
Can you explain why you want to prevent adding apps before world is initialized?

Reply:
Basically rules for a world are loaded (world is attached from CarpetServer) before the ScriptServer is reset, and the attachServer calls `set()` for most rules, if not all. Since rule apps are toggled from `set()`, that basically makes rule apps to "randomly" load or unload in that period, and not start if coming from setDefaults or from being their default value. Also for some reason it made rule apps that were loaded in that period from setDefault to be unusable until world reload without them enabled as default. That fixed it. Also that autoloaded apps are loaded in the same moment.
https://github.com/gnembon/fabric-carpet/blob/c827b041262dde181961c185dea4430f72bb2991/src/main/java/carpet/CarpetServer.java#L72-L78
https://github.com/gnembon/fabric-carpet/blob/a9e7bdce15ba825def2016b3f92869562e56c791/src/main/java/carpet/CarpetServer.java#L84-L88
(not sure if the name of that moment would really be `worldInitialized`, it's basically at `onServerLoadedWorlds` in CarpetServer)

Reply:
Wait. Actually that check may be completely useless, let me recheck.

Reply:
Yes, it is completely useless. My reasoning was to prevent it loading/unloading apps on SettingsManager reset, but totally forgot that the ScriptServer is killed _after_ that reset, not before, so the check and variable is pointless.

However, here there is still the discussion about whether rule apps should load at `onServerLoadedWorlds` (with autoloaded apps) or `onServerLoaded` (with regular rules)